libcontainer/standard_init_linux.go

```diff
@@ -189,14 +190,22 @@ func (l *linuxStandardInit) Init() error {
 			return err
 		}
 	}
-	if err := finalizeNamespace(l.config); err != nil {
+	if err := finalizeNamespace(l.config, l.addHome); err != nil {
 		return err
 	}
 	// finalizeNamespace can change user/group which clears the parent death
 	// signal, so we restore it here.
 	if err := pdeath.Restore(); err != nil {
 		return fmt.Errorf("can't restore pdeath signal: %w", err)
 	}
+
+	// In case we have any StartContainer hooks to run, and they don't
+	// have environment configured explicitly, make sure they will be run
+	// with the same environment as container's init.
```

Comment:
Maybe add a note that this behavior is not part of runtime-spec, but a de-facto historical thing we afraid to change.

Reply:
I think it worth to do it, because before this pr, it was doing in go-runtime, but now we have explicitly added the code to do this. We should let the reader know why we need this, and it may be removed in the future.